src/panels/Menu/TopMenuBar/TaskBarMenuChoices.tsx

```diff
@@ -0,0 +1,66 @@
+import { Dispatch, SetStateAction } from 'react';
+import { CheckboxIndicator, Menu } from '@mantine/core';
+
+import { ChevronRightIcon, TaskBarIcon } from '@/icons/icons';
+import { IconSize } from '@/types/enums';
+import { menuItemsData } from '@/windowmanagement/data/MenuItems';
+
+import { TopMenu } from './TopMenu';
+
+interface TaskBarChoicesProps {
+  visibleMenuItems: string[];
+  setVisibleMenuItems: Dispatch<SetStateAction<string[]>>;
+}
+
+export function TaskBarMenuChoices({
+  visibleMenuItems,
+  setVisibleMenuItems
+}: TaskBarChoicesProps) {
+  // Mapping menu item ids to boolean flag to determine their checked status
+  const checkedMenuItems: { [key: string]: boolean } = {};
+  visibleMenuItems.forEach((menuItemID) => {
+    checkedMenuItems[menuItemID] = true;
+  });
+
+  function toggleMenuItem(id: string): void {
+    setVisibleMenuItems((prevstate) => {
+      const index = prevstate.indexOf(id);
+      const isChecked = index >= 0;
```

Comment:
```suggestion
      const isChecked = index !== -1;
```
? Alternatively, 
```suggestion
      const isChecked = index > -1;
```

Either way, I think it's much clearer to compare with -1